node/blockchain/views/signed_change_request.py

```diff
@@ -1,15 +1,53 @@
+from rest_framework import status
+from rest_framework.exceptions import ValidationError as DRFValidationError
 from rest_framework.mixins import CreateModelMixin
+from rest_framework.response import Response
+from rest_framework.settings import api_settings
 from rest_framework.viewsets import GenericViewSet
 
-from node.blockchain.serializers.signed_change_request import SignedChangeRequestSerializer
+from node.blockchain.facade import BlockchainFacade
+from node.blockchain.models import Block
+from node.blockchain.serializers.signed_change_request import (
+    NodeDeclarationSignedChangeRequestSerializer, SignedChangeRequestSerializer
+)
+from node.core.exceptions import ValidationError
+from node.core.utils.cryptography import get_signing_key
+from node.core.utils.types import Type
 
 
 class SignedChangeRequestViewSet(CreateModelMixin, GenericViewSet):
     serializer_class = SignedChangeRequestSerializer
+    serializer_classes = {
+        Type.NODE_DECLARATION: NodeDeclarationSignedChangeRequestSerializer,
+    }
+
+    def get_serializer(self, *args, **kwargs):
+        serializer = super().get_serializer(*args, **kwargs)
+        serializer.is_valid(raise_exception=True)
+
+        signed_change_request_type = Type(serializer.validated_data['message']['type'])
+        serializer_class = self.serializer_classes.get(signed_change_request_type)
+        if not serializer_class:
+            raise ValidationError(f'Invalid signed_change_request_type: {signed_change_request_type}')
+
+        return serializer_class(*args, **kwargs)
 
     def perform_create(self, serializer):
-        instance = serializer.save()
+        signed_change_request = serializer.save()
+        blockchain_facade = BlockchainFacade.get_instance()
+        try:
+            Block.objects.add_block_from_signed_change_request(
+                signed_change_request=signed_change_request,
+                blockchain_facade=blockchain_facade,
+                signing_key=get_signing_key(),
+                validate=True
+            )
+        except ValidationError as ex:
+            raise DRFValidationError({api_settings.NON_FIELD_ERRORS_KEY: [str(ex)]})
 
-        # TODO(dmu) CRITICAL: Provide actual implementation of perform_create()
-        #                     https://thenewboston.atlassian.net/browse/BC-167
-        raise NotImplementedError(instance.json())
+    def create(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        self.perform_create(serializer)
+        headers = self.get_success_headers(serializer.data)
+        return Response(serializer.validated_data, status=status.HTTP_201_CREATED, headers=headers)
```

Comment:
why do we need to override create()? what is wrong original implementation of
```
return Response(serializer.data, status=status.HTTP_201_CREATED, headers=headers)
```

Reply:
When ViewSet use original create() method response json look wron
![response](https://user-images.githubusercontent.com/25705156/145265751-1468a47a-d49f-4b44-a4cc-e6e723013748.png)
g

Reply:
Should I find another way to return serialized data more correct?

Reply:
I didn't find how to avoid using of create() method. I added TODO to remove create() in the future. If you know now how to fix it please tell me.